concepts/ranges/introduction.md

```diff
@@ -6,8 +6,11 @@ They can be used for many things like quickly creating a collection, slicing str
 They are created using the range operator `..` or `...` (inclusive and exclusive respectively).
 
 ```ruby
-1..5 # => 1, 2, 3, 4, 5
-1...5 # => 1, 2, 3, 4
+1..5  # => 1..5
+1...5 # => 1...5
+
+(1..5).to_a # => [1, 2, 3, 4, 5]
+(1...5).to_a # => [1, 2, 3, 4]
```

Comment:
Oh, it also gives us an easy to verify side effect.  If we wanted to verify that the documentation is correct, it is easy to use `rubydoctest` to verify that it is still correct.

Reply:
That IRB syntax really isn't clear for me. It merging both the things you write/type and the things it's output in one format is very unclear to me. And I find it more unclear if we have multi-line things. e.g. if I want to do

```ruby
>> class Foobar
>>   def something
>>     5
>>   end
>> end
>> Foobar.new.something
=> 5
```

I just find seeing normal Ruby code without the `>>` prefix much much simpler to read:
```ruby
class Foobar
  def something
    5
  end
end

Foobar.new.something

# => 5
```

So to answer "Do we hate IRB in the documentation?", I can't speak for everyone, but I sure do!

I could maybe deal with the `irb>` syntax used in Rails docs, just on the lines that execute some actual code (see https://guides.rubyonrails.org/active_record_callbacks.html for examples) but I don't really like that either. I don't think its intuitive to someone who has no idea what irb is, and I don't think someone should need to know that to get through the track. I want people to be able to just type in what they see on the screen, not have to understand that `>>` has a special meaning and that they need to delete it if they copy + paste.

Reply:
Fair enough.  But then what is the `# => 5` at the end of that block?  Not very descriptive.  At least the IRB is a built in tool, and given with any Ruby install.

I am sure we are both biased, given our experience with the language and the tooling, which is why I am asking.

In "school" I explain the IRB and RI tooling that is included with Ruby.  When we are in "Practice" mode, then I kind of expect that experience to be there.  When in "Learn" mode, then I would expect the syntax  that we used to have been explained at some point.

Also, that is not output, it is the evaluation, the return, of either the line or the method.  Only lines that are `print` and `puts` gives output, if we consider it is something that we would see or that would be written to a file.

This is an extract from the guides link you gave.  Even this uses the proper IRB output and return distinctions.

And to me this behaves exactly as IRB, which is not surprising, given it is indicating it is coming from an IRB run:

```ruby
irb> User.new
You have initialized an object!
=> #<User id: nil>

irb> User.first
You have found an object!
You have initialized an object!
=> #<User id: 1>
```

The output is output, the evaluation shown as it shows there.  Less "interpretation" for the student if they see this and then run the Rails console, (aka, IRB), they are very likely to see exactly the same thing.

The shown block:

```ruby
>> class Foobar
>>   def something
>>     5
>>   end
>> end
>> Foobar.new.something
=> 5
```

… will not likely show what you would be typing anyway, so it may be confusing as well.

---

I do agree with the copy/paste being something that they should be able to do where appropriate without having to wrangle the lines of text.  Absolutely.

But doesn't `# => ` have a "special meaning" as well, since we have never (that I have found) explained what that means?

Reply:
> But doesn't # =>  have a "special meaning" as well, since we have never (that I have found) explained what that means?

`#` is a comment
` => ` is a signifier for output/evaluation/etc (as per the examples you pasted above).

So what the blocks in this PR are saying is "given this (runnable via copy+paste) code, here's a comment that shows what the output will be."

As you say, we definitely all have our biases and familiarities for this. I don't at all claim one is right or wrong, but I would like us continuing to use the syntax that I used on the existing concept exercises (e.g. see the tasks on https://exercism.org/tracks/ruby/exercises/lasagna/)

Reply:
@kotp It's worth noting that this is how official Ruby docs are written:
e.g. see the examples here: https://ruby-doc.org/core-2.7.0/Array.html#class-Array-label-Creating+Arrays